simuleval/evaluator/scorers/latency_scorer.py

```diff
@@ -272,6 +272,192 @@ def compute(
         return DAL
 
 
+@register_latency_scorer("ATD")
+class ATDScorer(LatencyScorer):
+    r"""
+    Average Token Delay (ATD) from
+    Average Token Delay: A Latency Metric for Simultaneous Translation
+    (https://arxiv.org/abs/2211.13173)
+
+    Different from speech segments, text tokens have no length
+    and multiple tokens can be output at the same time like subtitle.
+    Therefore, we set its length to be 0. However, to calculate latency in text-text,
+    we give virtual time 1 for the length of text tokens.
+
+    Usage:
+        ----latency-metrics ATD
+    """
+
+    def __call__(self, instances) -> float:
+        if isinstance(instances[0], TextInputInstance):
+            TGT_TOKEN_LEN = 1
+            SRC_TOKEN_LEN = 1
+            INPUT_TYPE = "text"
+            OUTPUT_TYPE = "text"
+        else:
+            SRC_TOKEN_LEN = 300  # 300ms per word
+            INPUT_TYPE = "speech"
+            if isinstance(instances[0], TextOutputInstance):
```

Comment:
The test failed because the scorer-only test only supports s2t now (we will update later.). So, for now, could you modify this line to something like this:

`if isinstance(instances[0], TextOutputInstance) or isinstance(instances[0], LogInstance):`

Sorry for the inconvience

Reply:
Sure, I have modified it.

Reply:
Hi, you might also need to import it 
```
from simuleval.evaluator.instance import TextInputInstance, TextOutputInstance, Instance, LogInstance
```

Reply:
Thank you for telling me. I have just imported it.

Reply:
Thanks! It's all checked! We will merge it very shortly

Reply:
Thank you! I'm glad.